TopSitesReport/XCTestCaseWaitExtension.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  XCTestCaseWaitExtension.swift
+//  DuckDuckGo
+//
+//  Copyright © 2017 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import XCTest
+
+extension XCTestCase {
+    
+    func wait(for duration: TimeInterval) {
+        let waitExpectation = expectation(description: "Wait expectation")
+        let when = DispatchTime.now() + duration
+        DispatchQueue.main.asyncAfter(deadline: when) {
+            waitExpectation.fulfill()
+        }
+        
+        // We use a large timeout buffer as jenkins stalls. This doesn't impact test
+        // speed as the expectation is fulfilled soon after the duration
+        let timeoutBuffer = 1800.0
+        waitForExpectations(timeout: duration + timeoutBuffer)
```

Comment:
Might be better to use a Semaphore here?  Expectations have the possibility of failure so this sends the wrong message to the reader as it'll never fail.

Reply:
I liked this idea but everything here is happening on the main thread. If we get the test to wait, the pageload itself also stalls, so when the semaphore signals from the bg thread there has been no page load progress.

I'll update this to make better use of expectations. I'm thinking we wait until page load + 2 seconds. That should increase the speed as well.

Reply:
Got you.  Try replacing the entire method with this:

`        RunLoop.main.run(until: Date(timeIntervalSinceNow: duration))`